src/openads/domain/position/Position.js

```diff
@@ -3,53 +3,40 @@ import {positionCreated} from './positionCreated'
 import {positionAlreadyDisplayed} from './positionAlreadyDisplayed'
 import {positionDisplayed} from './positionDisplayed'
 import DomainEventBus from '../service/DomainEventBus'
-import {positionSegmentationChanged} from './positionSegmentationChanged'
 import InvalidPositionStatusException from './InvalidPositionStatusException'
+import InvalidPositionSpecificationError from './InvalidPositionSpecificationError'
+import {positionUpdated} from './positionUpdated'
 
 export default class Position {
   /**
    * Create a new Position
    * @param {string} id
    * @param {string} name
-   * @param {string} source
-   * @param {string} placement
-   * @param {string} segmentation
-   * @param {Array<Array<>>}sizes
-   * @param {Object} native - Fields requested to the ad server
+   * @param {Object} specification
+   * @param {string} specification.source
```

Comment:
In the constructor, we don't receive specification.source like a parameter, it's necessary to specifically here?

Reply:
**specification.source** is later used in **checkSpecificationHasSource** function so it clarifies that source is required.

Reply:
Exactly :)